ampform/amp-form.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ampform
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const (
+	ERROR_CASE_AMP_FORM = "error"
+)
+
+func SubmitFormXHR(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("AMP-Access-Control-Allow-Source-Origin", buildSourceOrigin(r.Host))
+	w.Header().Set("Content-Type", "application/json")
+	if r.Method != "POST" {
+		http.Error(w, "post only", http.StatusMethodNotAllowed)
+	} else {
+		response := ""
+		name := r.FormValue("name")
+		if isUserTryingTheErrorDemo(name) {
+			w.WriteHeader(http.StatusBadRequest)
+		} else {
+			email := r.FormValue("email")
+			response = fmt.Sprintf("{\"name\":\"%s\", \"email\":\"%s\"}", name, email)
+		}
+		w.Write([]byte(response))
```

Comment:
because I am not printing the name (we removed the line which contained the name), so I don't need to put anything in the json.